### PR TITLE
Added OpInfo for nn.functional.dropout3d

### PR DESCRIPTION
**Pull request by fdrocha**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #81045

### CODE REVIEW

**Review by lezcano — COMMENTED** [review]
Good stuff. Just one small comment.

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -18285,7 +18285,31 @@ def error_inputs_mean(op_info, device, **kwargs):
...
+        dtypesIfCUDA=floating_types_and(torch.float16, torch.bfloat16),
+        skips=(
+            # lambda impl
+            DecorateInfo(unittest.expectedFailure, 'TestNormalizeOperators', 'test_normalize_operator_exhaustive'),
+            DecorateInfo(unittest.expectedFailure, 'TestJit', 'test_variant_consistency_jit'),
+            # torch.autograd.gradcheck.GradcheckError: While computing batched gradients, got:
+            # vmap: We do not yet support calling random operations inside of vmap.
+            # Please perform random operations outside of vmap as a workaround
+            DecorateInfo(unittest.expectedFailure, 'TestGradients', "test_forward_mode_AD"),
+            DecorateInfo(unittest.expectedFailure, 'TestGradients', "test_inplace_forward_mode_AD"),),
```
[the remark this answers is not in the document]
Comment (fdrocha) [reply torch/testing/_internal/common_methods_invocations.py:18303]: Both dropout2d and dropout3d already don't set it to False, they just XFAIL the corresponding tests.

Did you mean for dropout? 
I tried removing it there and adding the XFAILs, but something strange happens: it has an unexpected success when running on CUDA but still fails on CPU for test_forward_mode_AD_nn_functional_dropout. This only happens for dropout, in dropout2d and dropout3d it fails as expected on both cpu and cuda.

Not sure what to do here, do you think I should just XFAIL it for CPU?

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -18285,6 +18285,30 @@ def error_inputs_mean(op_info, device, **kwargs):
...
+        gradcheck_wrapper=wrapper_set_seed,
```
Comment (Lezcano) [inline torch/testing/_internal/common_methods_invocations.py:18304]: Is this necessary?

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lezcano) [reply torch/testing/_internal/common_methods_invocations.py:18303]: Sorry, I don't think I understand. What's the error that it throws if you set `check_batched_forward_grad=False`?

**Review by mruberry — COMMENTED** [review]
Hey @fdrocha! This looks really good overall. Once @Lezcano is happy I'll stamp it and then you can merge it using pytorchbot.

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fdrocha) [reply torch/testing/_internal/common_methods_invocations.py]: Sorry, I had misunderstood your first comment.
You are right, you can get rid of the XFAILS by using `check_batched_forward_grad=False`.
I added it here and also to dropout2d.

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (fdrocha) [reply torch/testing/_internal/common_methods_invocations.py]: No, it's not! I got rid of it here and also from dropout and dropout2d OpInfos

**Review by lezcano — APPROVED** [review]
LGTM! If the tests are green, I think this should be ready to be merged @mruberry

**Review by mruberry — APPROVED** [review]
Stamped! You can now merge this using pytorchbot, @fdrocha.

**Pull request closed without merging**